packages/block-renderer/src/components/block-renderer-provider.tsx

```diff
@@ -20,8 +22,16 @@ const useBlockRendererContext = (
 	useInlineStyles = false
 ) => {
 	const { data: settings } = useBlockRendererSettings( siteId, stylesheet, useInlineStyles );
-
-	const [ globalStyles ] = useSafeGlobalStylesOutput();
+	const { merged: config } = useContext( GlobalStylesContext );
+	// Remove section style variations until we handle them
+	if ( config?.styles?.blocks ) {
+		delete config.styles.blocks.variations;
+		delete config.styles.blocks[ 'core/group' ].variations;
+		delete config.styles.blocks[ 'core/column' ].variations;
+		delete config.styles.blocks[ 'core/columns' ].variations;
```

Comment:
How about traversing every block and removing its variations? Or do only these three blocks have section styles?

Reply:
[Just these three blocks at present](https://github.com/Automattic/themes/blob/5bc03e504bac0314c5f043b46f8c96c1efe6c5d6/assembler/theme.json#L792)

Reply:
I think `config.styles.blocks[ 'core/group' ]` and `config.styles.blocks[ 'core/columns' ]` don't exist and it throws error 🤔

Reply:
How about doing this at `mergeBaseAndUserConfigs`? I'm unsure but doing this here might make the `globalStyles` a new object on every render 🤔

Reply:
Let me try to push the changes

Reply:
I have not seen those errors but added a commit to protect the code.

> I'm unsure but doing this here might make the globalStyles a new object on every render 🤔
I don't notice any performance issues. I don't know why `globalStyles` would create a new object and why that can be a blocking issue.

Please feel free to contribute to the PR to fix this production bug.

Reply:
I have confirmed that the useMemo is not triggered on every render but three times only.

Reply:
Here it is: https://github.com/Automattic/wp-calypso/commit/ccbdd43acd44bbfd9103044db63ef60e4d26127d. Does it make sense?

Reply:
Looks good, thanks.
 
Can you clarify where `mergeBaseAndUserConfigs` is called and what is the goal of merging `base` and `user` to understand the impact of the change?

Reply:
The `mergeBaseAndUserConfigs` is called when we create the context for the `GlobalStylesContext.Provider`, so the `useGlobalStylesOutput` can generate the global styles from the `merged` config of the context.

The merged config is the **final** global styles for something like `useGlobalStylesOutput` to use.

Reply:
Nice!

For future reference, the global styles provider is connected to Assembler in https://github.com/Automattic/wp-calypso/blob/5c67e60d2650b8da90aaba166e7578bb7451f7a7/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx#L816 and merges the configs in https://github.com/Automattic/wp-calypso/blob/5c67e60d2650b8da90aaba166e7578bb7451f7a7/packages/global-styles/src/components/global-styles-provider/index.tsx#L58